### PR TITLE
docs: update python docs template

### LINKED ISSUE #570: Update sphinx conf.py for > 3.0.0
Update templates to incorporate change seen at https://github.com/googleapis/python-bigtable/pull/20

https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/docs/conf.py.j2
https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/noxfile.py.j2

### PR DESCRIPTION
**Pull request by kolea2**
Fixes #570

### CODE REVIEW

**Comment by kolea2** [comment]
@crwilcox PTAL

File: synthtool/gcp/templates/python_library/noxfile.py.j2 [hunk 1]
```diff
@@ -166,7 +166,7 @@ def docs(session):
     """Build the docs for this library."""
 
     session.install('-e', '.')
-    session.install('sphinx<3.0.0', 'alabaster', 'recommonmark')
+    session.install('sphinx', 'alabaster', 'recommonmark')
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/templates/python_library/noxfile.py.j2:169]: @crwilcox   IIRC, you needed a specific version of sphinx.  Will this break you?

Comment (kolea2) [reply synthtool/gcp/templates/python_library/noxfile.py.j2:169]: We made this change here: https://github.com/googleapis/synthtool/pull/576 and the docs build passed successfully, so we wanted to make this change more broadly.

File: synthtool/gcp/templates/python_library/docs/conf.py.j2 [hunk 2]
```diff
@@ -38,6 +38,7 @@ extensions = [
     "sphinx.ext.napoleon",
     "sphinx.ext.todo",
     "sphinx.ext.viewcode",
+    "recommonmark",
```
Comment (frankyn) [inline synthtool/gcp/templates/python_library/docs/conf.py.j2:41]: We did the same change in python-storage: https://github.com/googleapis/python-storage/pull/103

Didn't realize it came from this parent template. The rationale for it:

https://github.com/readthedocs/recommonmark

```
# for Sphinx-1.4 or newer
extensions = ['recommonmark'] # <---- this 

# for Sphinx-1.3
from recommonmark.parser import CommonMarkParser

source_parsers = {
    '.md': CommonMarkParser,
}

source_suffix = ['.rst', '.md']
```

File: synthtool/gcp/templates/python_library/noxfile.py.j2 (hunk 1 above)
Comment (crwilcox) [reply synthtool/gcp/templates/python_library/noxfile.py.j2:169]: @kolea2 is correct. A contributor made this change to bigtable specifically. And it seems @frankyn has had the same experience for storage specifically. This change undoes a mitigation while we were incompatible with newer versions of sphinx.

**Comment by kolea2** [comment]
@SurferJeffAtGoogle do you mind merging this? I don't have access :)

**Pull request merged**